# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsIScrollableFrame.h
# Commit: 065a45b8659d
# Full Hash: 065a45b8659dc25fb91738a7625dab916b76d0b6
# Author: Hiroyuki Ikezoe <hikezoe@mozilla.com>
# Date: 2019-06-27 09:34:48
# Regressor Bug: 1560237
# File Overlap Count: 1
# Description:
#   Bug 1560237 - Introduce nsIScrollableFrame::IsSmoothScroll. r=botond
#   
#   This is pretty much the same as ScrollStyles::IsSmoothScroll right now,
#   but in the next commit, we will no longer propagate scroll-behavior on <body> to
#   the root element so that nsIScrollableFrame::IsSmoothScroll will be changed
# ==============================================================================

diff -r 2b77a507ec05 -r 065a45b8659d layout/generic/nsIScrollableFrame.h
--- a/layout/generic/nsIScrollableFrame.h	Wed Jun 26 21:21:55 2019 +0000
+++ b/layout/generic/nsIScrollableFrame.h	Wed Jun 26 20:59:01 2019 +0000
@@ -564,6 +564,13 @@
   virtual bool SmoothScrollVisual(
       const nsPoint& aVisualViewportOffset,
       mozilla::layers::FrameMetrics::ScrollOffsetUpdateType aUpdateType) = 0;
+
+  /**
+   * Returns true if this scroll frame should perform smooth scroll with the
+   * given |aBehavior|.
+   */
+  virtual bool IsSmoothScroll(mozilla::dom::ScrollBehavior aBehavior =
+                                  mozilla::dom::ScrollBehavior::Auto) const = 0;
 };
 
 #endif
